daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ReinterpretTest.scala

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package engine
+
+import java.io.File
+
+import com.daml.lf.archive.{Decode, UniversalArchiveReader}
+import com.daml.bazeltools.BazelRunfiles
+import com.daml.lf.data.Ref._
+import com.daml.lf.data._
+import com.daml.lf.language.Ast._
+import com.daml.lf.transaction.{GlobalKeyWithMaintainers, NodeId, GenTransaction}
+import com.daml.lf.transaction.Node.{NodeRollback, NodeExercises}
+import com.daml.lf.value.Value._
+import com.daml.lf.command._
+import com.daml.lf.transaction.test.TransactionBuilder.assertAsVersionedValue
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.EitherValues
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.language.implicitConversions
+
+class ReinterpretTest
```

Comment:
Can we get some tests for the edge cases here, I’m particularly interested in uncatchable errors due to
1. version checks
2. because they are uncatchable in general, e.g. failed DAML-LF ensure, error call, ….

Reply:
failure ensure -- done
version check -- still to do
error call -- do we have an _uncatchable_ error primitive?

Reply:
Yes, Daml-LF’s error primitive is uncatchable. Daml’s `error` function does not use that primitive so you have to hand-write that to test it.
I think testing one uncatchable error is sufficient so either failed ensure (see my comment below, I think your test isn’t quite testing the right case), error call, ordering failure, ….

Reply:
Removed the previous useless ensure test.
And added a test which provokes an uncatchable error by creating a broken `Ord` instance (as we discussed),